bin/pit-init.sh

```diff
@@ -79,18 +84,43 @@ echo 'Generating Configuration ...'
     popd
 )
 
-SHASTA_CFG_DIR=$PREP_DIR/site-init
-if [ ! -d $SHASTA_CFG_DIR ] ; then
-    echo >&2 "Need $SHASTA_CFG_DIR; this needs to be created by hand if this is a first-time install."
+SITE_INIT=$PREP_DIR/site-init
+site_init_error=0
+if [ ! -d $SITE_INIT ] ; then
+    echo >&2 "Need $SITE_INIT; this needs to be created before invoking $0!"
     echo >&2 "See: https://github.com/Cray-HPE/docs-csm/blob/main/install/prepare_site_init.md#create-and-initialize-site-init-directory"
-    exit 1
+    site_init_error=1
+fi
+yq_error=0
+YQ_BINARY=/usr/bin/yq
+if [ -z ${CSM_PATH} ] ; then
+    echo >&2 "Can not find CSM tarball providing the yq binary, CSM_PATH is empty!"
+    yq_error=1
+elif [ ! -d ${CSM_PATH} ] ; then
+    echo >&2 "Can not find CSM_PATH: $CSM_PATH ; no such directory"
+    yq_error=1
+elif [ ! -f ${YQ_BINARY} ] ; then
+    YQ_BINARY="${CSM_PATH}/shasta-cfg/utils/bin/$(uname | awk '{print tolower($0)}')/yq"
```

Comment:
I htink we should check these paths at the beginning of the script and fail before we run `csi config init`.